src/compiler/program.ts

```diff
@@ -4229,6 +4229,16 @@ export function createProgram(rootNamesOrOptions: readonly string[] | CreateProg
             }
         }
 
+        if (options.isolatedDeclarations) {
+            if (options.out) {
+                createDiagnosticForOptionName(Diagnostics.Option_0_cannot_be_specified_with_option_1, "out", "isolatedDeclarations");

```

Comment:
...Is there a reason for this? It doesn't seem like anything in the implementation particularly cares about whether the output is concatenated by tsc or not. Or did I miss an error on all script-mode files somewhere and the reason for it?

Reply:
I think this was mostly for triple slash references, but we have since removed all of those anyway. Not sure what use case there is for this though, since the option is deprecated anyway and I don't think other tools what to implement the `out` behavior.

Reply:
I think @weswigham's question was more about `out`/`outFile` together, not specifically on `out`. `out` has been removed from 5.5 (the option stays behind in types so we can detect and error on it), but `outFile` has not.

Reply:
Yeah, that. It seems like the intent actually has nothing to do with `out`, and more script-mode globally scoped files. If that's the case, we should error on those, and not on `outFile` - amazingly, you can actually use `outFile` with `amd` modules. Eg,
```ts
// @declaration: true
// @strict: true
// @module: amd
// @outFile: out.js
// @filename: input1.ts
export class a {};

// @filename: input2.ts
import {a} from "./input1";
export const b = () => a;
```
and this actually produces declaration files that are widely useful, even if `amd` modules themselves are not.

Reply:
Removed the restriction.